lib/ansible/plugins/strategy/__init__.py

```diff
@@ -644,6 +645,21 @@ def _process_pending_results(self, iterator, one_pass=False, max_passes=None):
                     result_items = [task_result._result]
 
                 for result_item in result_items:
+                    if isinstance(original_task, RunBlock):
+                        # If this is a RunBlock task, we try and find the matching named block and
+                        # add that block to the iterator tasks for the given host
+                        target_name = original_task.args.get('name')
+                        for nb in original_task.play.named_blocks:
+                            if nb.name == target_name:
+                                nb_copy = nb.copy()
+                                nb_copy.parent = original_task
+                                nb_copy.vars.update(original_task.vars)
```

Comment:
Not sure if there should be a deep copy of task vars here or not.

Reply:
it should be pretty safe to use the vars as is